allennlp/commands/evaluate.py

```diff
@@ -123,47 +138,68 @@ def evaluate_from_args(args: argparse.Namespace) -> Dict[str, Any]:
         cuda_device=args.cuda_device,
         overrides=args.overrides,
     )
-    config = archive.config
+    config = deepcopy(archive.config)
     prepare_environment(config)
     model = archive.model
     model.eval()
 
     # Load the evaluation data
-
     dataset_reader = archive.validation_dataset_reader
 
-    evaluation_data_path = args.input_file
-    logger.info("Reading evaluation data from %s", evaluation_data_path)
-
-    data_loader_params = config.pop("validation_data_loader", None)
-    if data_loader_params is None:
-        data_loader_params = config.pop("data_loader")
-    if args.batch_size:
-        data_loader_params["batch_size"] = args.batch_size
-    data_loader = DataLoader.from_params(
-        params=data_loader_params, reader=dataset_reader, data_path=evaluation_data_path
-    )
-
-    embedding_sources = (
-        json.loads(args.embedding_sources_mapping) if args.embedding_sources_mapping else {}
-    )
+    # split files
+    evaluation_data_path_list = args.input_file.split(":")
+    if args.output_file is not None:
+        output_file_list = args.output_file.split(":")
+        assert len(output_file_list) == len(
+            evaluation_data_path_list
+        ), "number of output path must be equal number of dataset "
+    if args.predictions_output_file is not None:
+        predictions_output_file_list = args.predictions_output_file.split(":")
+        assert len(predictions_output_file_list) == len(
+            evaluation_data_path_list
+        ), "number of predictions_output_file path must be equal number of dataset "
+
+    # output file
+    output_file_path = None
+    predictions_output_file_path = None
+
+    for index in range(len(evaluation_data_path_list)):
+        config = deepcopy(archive.config)
```

Comment:
Do we modify the config anywhere? Why do we need to `deepcopy`?

Reply:
The config file is allennlp's [Params](https://docs.allennlp.org/main/api/common/params/#params).  Everytime, we get or pop a value from the Params object, the code is forced to the Params. I'm not sure, if there's a cleaner way around this though.